### PR TITLE
Adding descriptive language to tests and allowing usage of .env

### LINKED ISSUE #2886: Make tests descriptive
Currently, test failures don't tell you a great deal.

We should ensure that tests accurately describe the problem on failure, using the `$message` param in PHPUnit's `assert` methods.

### LINKED ISSUE #2893: Tests fail locally for me
## While trying to do this:

Run local unit tests using PHPUnit 9.5.2.

## I encountered this error:

8 of them, actually! I'm not sure why these are failing locally for me yet (partially because tests don't have English descriptions per #2886).

```
There were 8 failures:

1) Tests\API\BasicAPITest::testConnection
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/API/BasicAPITest.php:24

2) Tests\API\BasicAPITest::testAuthenticated
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/API/BasicAPITest.php:46

3) Tests\API\UploadTest::testUpload
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/API/UploadTest.php:32

4) Tests\Core\GatekeeperTest::testGatekeeper
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/Core/GatekeeperTest.php:17

5) Tests\Core\GatekeeperTest::testAdminGatekeeper
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/Core/GatekeeperTest.php:41

6) Tests\Core\PermalinkStructureTest::testPermalinks
Failed asserting that false is not false.

/Users/ben/Sites/known/public/Tests/Core/PermalinkStructureTest.php:34

7) Tests\Pages\HomepageTest::testHomepageLoads
Failed asserting that a boolean is not empty.

/Users/ben/Sites/known/public/Tests/Pages/HomepageTest.php:18

8) Tests\Pages\RSSTest::testValid
Failed asserting that false is true.

/Users/ben/Sites/known/public/Tests/Pages/RSSTest.php:24
```

## Some other notes:

This test run is on the very latest dev code as of the time of posting.

### PR DESCRIPTION
**Pull request by benwerd**
## Here's what I fixed or added:

All tests now have text descriptions for their assertions. In addition, the testing environment now allows the use of .env environment files for development, which make per-install environment variables much easier to handle. (This allows a user to have multiple installations of Known with different environment vars without needing to do anything on the command line.)

Fixes #2886 
Fixes #2893 

## Here's why I did it:

Test results were confusing, and environment variable setting wasn't clear. `.env` files are the most common solution for this problem in development.

### CODE REVIEW

File: Tests/API/BasicAPITest.php [hunk 1]
```diff
@@ -42,10 +50,10 @@ public function testAuthenticated()
             $content = json_decode($result['content']);
             $response = $result['response'];
 
-            $this->assertTrue(empty($result['error']));
-            $this->assertTrue(!empty($content));
-            $this->assertTrue(!empty($content->location));
-            $this->assertTrue($response == 200);
+            $this->assertTrue(empty($result['error']), 'The result should not contain an error property.');
```
Comment (lindner) [inline Tests/API/BasicAPITest.php]: I think there's an assertEmpty() and an assertArrayHasKey which might help with the error messages.

File: Tests/API/BasicAPITest.php [hunk 2]
```diff
@@ -42,10 +50,10 @@ public function testAuthenticated()
...
+            $this->assertTrue(!empty($content), 'Retrieved content should not be empty. Have you set the KNOWN_DOMAIN environment variable? Endpoint: ' . $endpoint);
+            $this->assertTrue(!empty($content->location), 'Response should contain the location of the post.');
+            $this->assertTrue($response == 200, 'The response should have returned a 200 HTTP response.');
```
Comment (lindner) [inline Tests/API/BasicAPITest.php]: assertEquals

File: Tests/Data/DataConciergeTest.php [hunk 3]
```diff
@@ -168,8 +168,8 @@ public function testGetByRangeNoResults()
             $search['rangeVariable']['$gt'] = 'c';
 
             $count = \Idno\Entities\GenericDataItem::countFromX('Idno\Entities\GenericDataItem', $search);
-            $this->assertTrue(is_int($count));
-            $this->assertTrue($count == 0);
+            $this->assertTrue(is_int($count), 'A count of entities should be an integer.');
+            $this->assertTrue($count == 0, 'No entities should match our query.');
```
Comment (lindner) [inline Tests/Data/DataConciergeTest.php]: assertEquals here too.  etc.

**Review by lindner — APPROVED** [review]
Overall LGTM and approval in advance but there are CI failures.  And better use of assertions can help with the error messages.

**Comment by benwerd** [comment]
Thanks @lindner! Good pointers.

**Pull request merged**